galley/pkg/config/source/kube/inmemory/kubesource.go

```diff
@@ -303,10 +304,67 @@ func (s *KubeSource) parseChunk(r *collection.Schemas, name string, lineNum int,
 		return kubeResource{}, err
 	}
 
+	// Build flat map for analyzers if the line JSON object exists, if the YAML text is ill-formed, this will be nil
+	fieldMap := make(map[string]int)
+
+	// yamlv3.Node contains information like line number of the node, which will be used with its name to construct the field map
+	yamlChunkNode := yamlv3.Node{}
+	err = yamlv3.Unmarshal(yamlChunk, &yamlChunkNode)
+	if err == nil && len(yamlChunkNode.Content) == 1 {
+
+		// Get the Node that contains all the YAML chunk information
+		yamlNode := yamlChunkNode.Content[0]
+
+		BuildFieldPathMap(yamlNode, lineNum, "", fieldMap)
+	}
+
 	pos := rt.Position{Filename: name, Line: lineNum}
 	return kubeResource{
 		schema:   schema,
 		sha:      sha1.Sum(yamlChunk),
-		resource: rt.ToResource(objMeta, schema, item, &pos),
+		resource: rt.ToResource(objMeta, schema, item, &pos, fieldMap),
 	}, nil
 }
+
+// BuildFieldPathMap builds the flat map for each field of the YAML resource
+func BuildFieldPathMap(yamlElementNode *yamlv3.Node, startLineNum int, curPath string, fieldPathMap map[string]int) {
```

Comment:
I think it doesn't. Only after parsing the YAML chunk correctly and before converting all of the objects to the resource will require this function to build the map. Or if someone else in other places needs to use this function to build a map, then it can be exported.